### PR TITLE
[AAE-19968] Field visibility condition using Integer and greater than not working

### PR DESCRIPTION
**Pull request by wiktord2000**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-19968

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/form/services/widget-visibility.service.spec.ts [hunk 1]
```diff
@@ -421,6 +421,30 @@ describe('WidgetVisibilityService', () => {
             expect(rightValue).toBe('dropdown_label');
         });
 
+        it('should getRightValue be able to return 0 when rightType is set to value', () => {
```
Comment (eromano) [inline lib/core/src/lib/form/services/widget-visibility.service.spec.ts:424]: I guess we need a unit test that check how the function behave in case rightValue is null after this change

**Review by eromano — CHANGES_REQUESTED** [review]

File: lib/core/src/lib/form/models/widget-visibility.model.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ export class WidgetVisibilityModel {
     }
 
     get rightValue(): any {
-        if (this.json.rightValue) {
+        if (this.json.rightValue != null) {
```
Comment (tomgny) [inline lib/core/src/lib/form/models/widget-visibility.model.ts:89]: shouldn't we use here strict (`!==`) checking? with this implementation, if rightValue is undefind, the condition will be met, is this intended?

File: lib/core/src/lib/form/services/widget-visibility.service.spec.ts (hunk 1 above)
Comment (mauriziovitale) [inline lib/core/src/lib/form/services/widget-visibility.service.spec.ts:424]: rightType is set to value 
it should return 0 when rightValue is a number
it should return false when rightValue is a boolean
it should return empty string when rightValue is a string
it should return X when rightValue is undefined or null
it should return Y when rightValue is an object. ??

Comment (mauriziovitale) [reply lib/core/src/lib/form/services/widget-visibility.service.spec.ts:424]: I think we also need to review the same code on left 

  ```
  get leftValue(): any {
        if (this.json.leftValue) {
  ```      
      
I would better check the scenario because it looks inconsistent. We are setting the **rightType** to **value** and then the **value** is null ? Why is it null?

File: lib/core/src/lib/form/models/widget-visibility.model.ts (hunk 2 above)
Comment (wiktord2000) [reply lib/core/src/lib/form/models/widget-visibility.model.ts:89]: It was intended but not sure whether the best solution

File: lib/core/src/lib/form/services/widget-visibility.service.spec.ts [hunk 3]
```diff
@@ -421,6 +421,51 @@ describe('WidgetVisibilityService', () => {
             expect(rightValue).toBe('dropdown_label');
         });
 
+        describe('getRightValue when rightType is set to value', () => {
+            beforeEach(() => {
+                visibilityObjTest.rightType = 'value';
+            });
+
+            it('should be able to return 0 when rightValue is a number', () => {
```
Comment (wiktord2000) [inline lib/core/src/lib/form/services/widget-visibility.service.spec.ts:429]: add. test for e.g. 10 > 0

File: lib/core/src/lib/form/services/widget-visibility.service.spec.ts [hunk 4]
```diff
@@ -421,6 +421,51 @@ describe('WidgetVisibilityService', () => {
...
+                visibilityObjTest.rightValue = 0;
+                const rightValue: any = service.getRightValue(formTest, visibilityObjTest);
+
+                expect(rightValue).toBe(0);
+            });
+
+            it('should be able to return false when rightValue is a boolean', () => {
```
Comment (wiktord2000) [inline lib/core/src/lib/form/services/widget-visibility.service.spec.ts:436]: Check true > "false"

**Review by mauriziovitale — APPROVED** [review]
